### ISSUE TITLE
Missing octects file import after 0.6.1 tag released

### ISSUE THREAD

**Issue by WibblyGhost** [pyasn1/pyasn1#76]
Pyasn1 v 0.6.1 seems to have broken imports for the DgramAsyncioPorotocol used in pysnmp and cannot run pysnmp code due to an improper import. Using python 3.12 currently.

```python
File "features/steps/snmp_steps.py", line 8, in <module>
    from features.classes.snmp_classes import SnmpClient, SnmpResult
  File "/home/taitnet/Programming/integration-tests/features/classes/snmp_classes.py", line 7, in <module>
    from pysnmp.carrier.asyncio.dgram.udp import domainName, UdpAsyncioTransport
  File "/home/taitnet/Programming/integration-tests/.venv3/lib/python3.12/site-packages/pysnmp/carrier/asyncio/dgram/udp.py", line 35, in <module>
    from pysnmp.carrier.asyncio.dgram.base import DgramAsyncioProtocol
  File "/home/taitnet/Programming/integration-tests/.venv3/lib/python3.12/site-packages/pysnmp/carrier/asyncio/dgram/base.py", line 43, in <module>
    from pysnmp import debug
  File "/home/taitnet/Programming/integration-tests/.venv3/lib/python3.12/site-packages/pysnmp/debug.py", line 8, in <module>
    from pyasn1.compat.octets import octs2ints
ModuleNotFoundError: No module named 'pyasn1.compat.octets'
```

Tracking the commits down to this hash which removed the octects file from the repo, but missed the import in the file `.venv3/lib/python3.12/site-packages/pysnmp/debug.py`.
Problematic commit: https://github.com/pyasn1/pyasn1/commit/6f770ba886a8931c35cb090a5c3a6d67f5a41bd9

the code is used for a hexdump object of sorts.
```python
def hexdump(octets):
    return " ".join(
        [
            "{}{:02X}".format(n % 16 == 0 and ("\n%.5d: " % n) or "", x)
            for n, x in zip(range(len(octets)), octs2ints(octets))
        ]
    )
```

**Comment by cpackham-atlnz** [comment]
I reported an issue on pysnmp for this https://github.com/lextudio/pysnmp/issues/113

**Comment by lextm** [comment]
Depending on the PYSNMP release you are using, new releases on the upgrade path have been out. So, this issue can be closed for PySNMP.

**Comment by jonathanslenders** [comment]
It also looks like no wheels were released for this release.

**Comment by droideck** [comment]
> It also looks like no wheels were released for this release.

With the `python -m build` migration, it was missed. Thanks! I added it now.

As for this issue, yes, pysnmp should stop using Python 2-related code, too, when used with 0.6.x version (but mainly with 0.6.1 as we cleaned up the rest of deprecated Python 2 code)

**Comment by WibblyGhost** [comment]
Thanks for raising a issue on PySNMP, I'll close this issue and track it on the other repo.

**Comment by onealj** [comment]
Because the removal/move of octets was a breaking change, these changes to pyasn1 probably should have been released as 0.7.0 instead of 0.6.1.

My project uses `PySNMP==4.4.12`, which declares `pyasn1>=0.2.3` in its requirements.txt. Adding `pyasn1<=0.6.0` to my project's dependencies resolves the `pyasn1.compat.octets` problem without forcing me to upgrade to the latest version of pysnmp and pyasn1.

**Comment by droideck** [comment]
> Because the removal/move of pyasn1.compat.octets was a breaking change, these changes to pyasn1 maybe should have been released as 0.7.0 instead of 0.6.1. Though that wouldn't have avoided import errors for other projects that didn't pin or specify a maximum version of pyasn1.

The 0.6.0 version removed Python 2 support. 0.6.1 just cleaned up Python 2-related code that was missed during the 0.6.0 release.

If you want to use the Python 2 supported version, you should set `pyasn1<0.6.0` - this way, you'll avoid any issue with unsupported Python 2 code.

**Comment by lextm** [comment]
@onealj Such breaking changes are welcome because they force users to take immediate actions.

As new maintainers of PySNMP, we urge users to upgrade to 6.x/7.x.
